frontend/packages/console-shared/src/hooks/useUserSettingsLocalStorage.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+// FIXME upgrading redux types is causing many errors at this time
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore
+import { useSelector } from 'react-redux';
+import { RootState } from '@console/internal/redux';
+import { deseralizeData, seralizeData } from '../utils/user-settings';
+
+const CONFIGMAP_LS_KEY = 'console-user-settings';
+
+export const useUserSettingsLocalStorage = <T>(
+  key: string,
+  defaultValue: T,
+  watch: boolean = true,
+): [T, React.Dispatch<React.SetStateAction<T>>] => {
+  const keyRef = React.useRef(key);
+  const defaultValueRef = React.useRef(defaultValue);
+  const userUid = useSelector(
+    (state: RootState) => state.UI.get('user')?.metadata?.uid ?? 'kubeadmin',
+  );
+  const storageConfigNameRef = React.useRef(`${CONFIGMAP_LS_KEY}-${userUid}`);
+  const [lsData, setLsData] = React.useState(() => {
+    const valueInLocalStorage = deseralizeData(localStorage.getItem(storageConfigNameRef.current));
+    return valueInLocalStorage?.[keyRef.current]
+      ? valueInLocalStorage[keyRef.current]
+      : defaultValueRef.current;
+  });
+  const lsDataRef = React.useRef<T>(lsData);
+
+  const localStorageUpdated = React.useCallback(
+    (ev: StorageEvent) => {
+      if (ev.key === storageConfigNameRef.current) {
+        const lsConfigMapData = deseralizeData(localStorage.getItem(storageConfigNameRef.current));
+        if (
+          lsData !== undefined &&
+          lsConfigMapData?.[keyRef.current] &&
+          seralizeData(lsConfigMapData[keyRef.current]) !== seralizeData(lsData)
+        ) {
+          setLsData(lsConfigMapData[keyRef.current]);
```

Comment:
Shouldn't this be deserialized?

Reply:
in case of localStorage we are seralizing and deseralizing complete data in Configmap key so din't need to do it

```
const lsConfigMapData = deseralizeData(localStorage.getItem(storageConfigNameRef.current));
```